src/features/hotspots/setup/HotspotSetupWifiConnectingScreen.tsx

```diff
@@ -64,23 +68,45 @@ const HotspotSetupWifiConnectingScreen = () => {
   }, [addGatewayTxn, hotspotAddress, hotspotType, navigation])
 
   const connectToWifi = useCallback(async () => {
+    // Segment track for wifi connection
+    track(HotspotEvents.WIFI_CONNECTION_STARTED, {
+      network,
+    })
+
     const response = await setWifi(network, password)
     if (response === 'not_found') {
+      // Segment track for wifi connection
+      track(HotspotEvents.WIFI_CONNECTION_FAILED, {
+        network,
+        message: 'Something went wrong.',
+      })
+
       showOKAlert({
         titleKey: 'generic.error',
         messageKey: 'generic.something_went_wrong',
       })
       navigation.goBack()
     } else if (response === 'invalid') {
+      // Segment track for wifi connection
+      track(HotspotEvents.WIFI_CONNECTION_FAILED, {
+        network,
+        message: 'Invalid password.',
+      })
+
       showOKAlert({
         titleKey: 'generic.error',
         messageKey: 'generic.invalid_password',
       })
       navigation.goBack()
     } else {
+      // Segment track for wifi connection
+      track(HotspotEvents.WIFI_CONNECTION_FINISHED, {
+        network,
```

Comment:
Don't think we should track SSIDs. Is that what `network` is?

Reply:
It's SSID. Let me remove it.